src/StreamSnapshotProjection.php

```diff
@@ -31,15 +31,16 @@ public function __construct(ReadModelProjector $readModelProjector, string $stre
     {
         $this->readModelProjector = $readModelProjector;
         $this->streamName = $streamName;
-    }
 
-    public function __invoke(bool $keepRunning = true)
-    {
         $this->readModelProjector
             ->fromStream($this->streamName)
             ->whenAny(function ($state, Message $event): void {
                 $this->readModel()->stack('replay', $event);
-            })
-            ->run($keepRunning);
+            });
+    }
+
+    public function __invoke(bool $keepRunning = true)
```

Comment:
I'm also curious about why using `__invoke` here?

By the way, the doc is wrong then (it mentions a `run()` method which does not exist: https://github.com/prooph/snapshotter/blame/master/docs/snapshots.md#L53)

Reply:
updated the docs